src/test/java/nextstep/utils/GithubTestController.java

```diff
@@ -1,23 +1,117 @@
 package nextstep.utils;
 
+import nextstep.auth.token.oauth2.github.GithubAccessTokenRequest;
 import nextstep.auth.token.oauth2.github.GithubAccessTokenResponse;
 import nextstep.auth.token.oauth2.github.GithubProfileResponse;
+import org.springframework.context.annotation.Profile;
 import org.springframework.http.ResponseEntity;
-import org.springframework.web.bind.annotation.GetMapping;
-import org.springframework.web.bind.annotation.PostMapping;
-import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.NoSuchElementException;
+
+import static nextstep.utils.GithubTestController.GithubResponses.*;
 
 @RestController
+@Profile("test")
 public class GithubTestController {
 
     @PostMapping("/github/login/oauth/access_token")
-    public ResponseEntity<GithubAccessTokenResponse> accessToken() {
-        return ResponseEntity.ok().build();
+    public ResponseEntity<GithubAccessTokenResponse> accessToken(@RequestBody GithubAccessTokenRequest githubAccessTokenRequest) {
+        String code = githubAccessTokenRequest.getCode();
+        GithubResponses githubResponse = getGithubResponseWithCode(code);
+        GithubAccessTokenResponse result = new GithubAccessTokenResponse();
+
+        switch (githubResponse) {
+            case 사용자1:
+                result.setAccessToken(사용자1.getAccessToken());
+                break;
+            case 사용자2:
+                result.setAccessToken(사용자2.getAccessToken());
+                break;
+            case 사용자3:
+                result.setAccessToken(사용자3.getAccessToken());
+                break;
+            case 사용자4:
+                result.setAccessToken(사용자4.getAccessToken());
+                break;
+            default:
+                throw new IllegalStateException("Unexpected value: " + githubResponse);
+        }
```

Comment:
GithubResponses가 enum 클래스이니 swtich 분기를 하지 않아도 될 것 같은데 어떻게 생각하시나요?
```suggestion
        result.setAccessToken(githubResponse.getAccessToken());
```